### ISSUE TITLE
Analyzer fails to promote target of null-aware call to getter

### ISSUE THREAD

**Issue by stereotype441** [dart-lang/sdk#46937]
(Based on https://github.com/flutter/flutter/issues/88023)

The following program is treated differently by the analyzer and the CFE:
```dart
abstract class C {
  void Function(C c) get functionGetter;
  void method(C c);
}

void f(C? c) {
  c?.functionGetter(c); // (1)
}

void g(C? c) {
  c?.method(c);
}

main() {}
```
The analyzer considers there to be an error at the line marked (1): `The argument type 'C?' can't be assigned to the parameter type 'C'.`.  It considers the line marked (2) to be ok.

The CFE considers the program to have no errors.

I believe the CFE's interpretation is correct.

My best guess as to what's happening is that when the analyzer notices that `c?.functionGetter(c)` resolves to a getter (rather than a method), and rewrites the AST into an invocation of a property rather than a method call, it somehow loses track of the fact that null shorting is in effect, so it thinks that the argument `c` might be nullable.

I will be on vacation for the rest of the week but I'll be happy to take a look at this on Monday.

**Comment by scheglov** [comment]
https://dart-review.googlesource.com/c/sdk/+/210943

**Comment by stereotype441** [comment]
Fixed by 66b3b1a6c966932938dd7434a51be706b9640f1a.
